packages/core/src/reporters/json-reporter.ts

```diff
@@ -0,0 +1,62 @@
+import * as path from 'path';
+
+import { StrykerOptions } from '@stryker-mutator/api/core';
+import { Logger } from '@stryker-mutator/api/logging';
+import { commonTokens, tokens } from '@stryker-mutator/api/plugin';
+import { mutationTestReportSchema, Reporter } from '@stryker-mutator/api/report';
+
+import fileUrl = require('file-url');
+
+import * as HtmlReporterUtil from './html/html-reporter-util';
+
+const DEFAULT_BASE_FOLDER = path.normalize('reports/mutation');
+const DEFAULT_INDENTION = 0;
+export const RESOURCES_DIR_NAME = 'strykerResources';
+
+export default class JsonReporter implements Reporter {
+  private _baseDir!: string;
+  private mainPromise: Promise<void> | undefined;
+
+  constructor(private readonly options: StrykerOptions, private readonly log: Logger) {}
+
+  public static readonly inject = tokens(commonTokens.options, commonTokens.logger);
+
+  public onMutationTestReportReady(report: mutationTestReportSchema.MutationTestResult) {
+    this.mainPromise = this.generateReport(report);
+  }
+
+  public wrapUp() {
+    return this.mainPromise;
+  }
+
+  private async generateReport(report: mutationTestReportSchema.MutationTestResult) {
+    const jsonReportFile = path.resolve(this.baseDir, 'mutation.json');
+    await this.cleanBaseFolder();
+    let indent = DEFAULT_INDENTION;
+    if (this.options.jsonReporter && this.options.jsonReporter.spacing) {
+      indent = this.options.jsonReporter.spacing;
```

Comment:
In hind sight, I think this feature is a bit expensive for what it does. Expensive in the sense that we need to document and maintain it. In practise, the Json reporter would not be used to show the report to a human (i.e. open in an editor).

Is there a specific reason you would want this feature?

Reply:
Do you only mean the indention option or the whole json reporter in general? I think that a json reporter should definitely be supported since there is currently no reporter for easy data exchange. If you mean the indention option only, you are probably right. I just added it because it was a very easy addition and is probably wanted. :)

Reply:
I've now removed the indention option and use the default value (0 = no indention) as a static constant.